benchmarks/vector_db_compare/rag_qdrant_runner.py

```
@@ -0,0 +1,210 @@
+    ordered = sorted(values); return ordered[max(0, math.ceil(q * len(ordered)) - 1)]
+def quality(ids, parents, chunks, relevant_parents):
+    relevant, relevantp = set(chunks), set(relevant_parents)
+    precision = lambda k: sum(value in relevant for value in ids[:k]) / k
+    recall = lambda k: sum(value in relevant for value in ids[:k]) / len(relevant)
+    def ndcg(k):
+        actual = sum(1 / math.log2(i + 2) for i, value in enumerate(ids[:k]) if value in relevant)
+        ideal = sum(1 / math.log2(i + 2) for i in range(min(k, len(relevant))))
+        return actual / ideal
+    rank = next((i + 1 for i, value in enumerate(ids[:10]) if value in relevant), 0)
+    parent_recall = lambda k: len({value for value in parents[:k] if value in relevantp}) / len(relevantp)
+    achievable = lambda relevant_count, k: min(relevant_count, k) / relevant_count
```

Ruff (0.16.3)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `achievable` as a `def`

(E731)